eng/versioning.targets

```diff
@@ -155,4 +155,14 @@
       WriteOnlyWhenDifferent="true" />
 
   </Target>
+
+  <!-- Removes specified assembly level attributes. https://github.com/dotnet/runtime/issues/44257-->
+  <Target Name="RemoveSupportedPlatformAssemblyLevelAttribute"
+          AfterTargets="GetAssemblyAttributes">
+    <ItemGroup>
+      <AssemblyAttribute Remove="System.Runtime.Versioning.SupportedOSPlatformAttribute" />
```

Comment:
@Anipik isn't it will remove the attribute added above on row 27? https://github.com/dotnet/runtime/pull/43965/files#diff-3e71d76d1049d3878247187d660b7cab477a963020055effc4160ac4e85962cfR26-R30

Reply:
@Anipik or @buyaa-n, are you following-up on the break? The fix would probably to add the SupportedOSPlatformAttribute in a target after this one runs.

Reply:
Just checked my build dlls, don't see it is removed from the build, but i might need to do a clean build to make sure, if the above rule for windows builds runs after this one i think it will be OK, i will follow up after cleaning my repo and doing new build

Reply:
We discussed this offline and we don't believe this needs reaction. The attributes that are removed aren't the same as the one that is added.

Reply:
@ViktorHofer seems you are misunderstood here i meant about the rule adding Supported on windows https://github.com/dotnet/runtime/blob/f56a56f90ff9124c85e4d889faeeca0824d2d437/eng/versioning.targets#L25-L30

We talked offline about the rule adding Unsupported on browser which wouldn't be affected with this change
https://github.com/dotnet/runtime/blob/f56a56f90ff9124c85e4d889faeeca0824d2d437/eng/versioning.targets#L36-L42

Anyways as i mentioned before i still see the attribute added from my local build result, i will check it again when i do clean build

Reply:
You are right, I overlook the `SupportedOSPlatform` define above the `UnsupportedOSPlatform` one. Even though this is ugly, nothing is broken by this as the item `SupportedOSPlatform` that we define isn't equal to the `SupportedOSPlatformAttribute` item, so msbuild doesn't remove it.

Reply:
We should clean this up though.

Reply:
> nothing is broken by this as the item SupportedOSPlatform that we define isn't equal to the SupportedOSPlatformAttribute item, so msbuild doesn't remove it.

Oh that's why, yes we hope we will have better solution with https://github.com/dotnet/runtime/issues/44257